File: 3DRadSpace/Engine3DRadSpace/IObject2D.cpp
#include "IObject2D.hpp"

using namespace Engine3DRadSpace;
using namespace Engine3DRadSpace::Math;

IObject2D::IObject2D(const std::string &name, const std::string &tag, bool enabled, bool visible, const Vector2&pos, const
                     Vector2&scale, 
                     float rotation, const Vector2&pivot, float depth) :
	IObject(name, tag, false, false),
	Position(pos),
	Rotation(rotation),
	RotationCenter(pivot),
	Scale(scale),
	Depth(depth)
{
}

IObject2D::IObject2D(Game *game, const std::string &name, const std::string &tag, bool enabled, bool visible, const Vector2 &pos,
                     const Vector2& scale, float rotation, const Vector2&pivot, float depth): 
	IObject(game, name, tag, false, false),
	Position(pos),
	Rotation(rotation),
	RotationCenter(pivot),
	Scale(scale),
	Depth(depth)
{
}

File: 3DRadSpace/3DRadSpace_CSharp/Objects/IObject2D.cs
using System.Numerics;

namespace Engine3DRadSpace.Objects;

public interface IObject2D : IObject, IDrawable2D
{
	public Vector2 Position { get; set; }
	public Vector2 RotationCenter { get; set; }
	public Vector2 Scale { get; set; }
    /// <summary>
    /// Z-depth for layering 2D objects.
    /// </summary>
    public float Depth { get; set; }
	
	/// <summary>
	/// Rotation in radians.
	/// </summary>
	public float Rotation { get; set; }
}
